src/const.ts

```diff
@@ -11,3 +11,77 @@ export interface Manifest {
   name: string;
   builds: Build[];
 }
+
+interface BaseFlashState {
+  state: State;
+  message: string;
+  manifest?: Manifest;
+  chipFamily?: "ESP32" | "ESP8266" | "ESP32-S2" | "Unknown Chip";
+}
+
+export interface InitializingState extends BaseFlashState {
+  state: State.INITIALIZING;
+  details: { done: boolean };
+}
+
+export interface ManifestState extends BaseFlashState {
+  state: State.MANIFEST;
+  details: { done: boolean };
```

Comment:
I think that we can just flatten the details into the state object. `{ state, done }`. Also `done` is not necessary because `done = state == "finished" || state == "error"`

Reply:
Done is for the manifest step, otherwise without the message, you would not know if the step is starting or finished.